docs/team/repository-code-standards.md

```diff
@@ -0,0 +1,36 @@
+# Repository Management and Code Standards
+
+JupyterHub projects should take advantage of pre-commit and tools to help maintain consistent formatting within a repo to improve overall code quality, review efficiency, and readability of code.
+Sensible use of pre-commit to run formatting tools and code linters can add consistency and improve maintainability.
+
+## Preferred tools and pre-commit hooks
+
+The team has found the following tools and their pre-commit hooks to be useful. The following pre-commit hooks can be added to any repo when convenient, though please communicate with anyone who has an open pull request if it will lead to major conflicts:
+
+  - [black](https://black.readthedocs.io/)
+  - [prettier](https://prettier.io/)
+  - [flake8](https://flake8.pycqa.org/en/latest/)
+  - **TODO: standard pre-commit hooks, such as isort, etc. (To be determined and listed here)**
+
+## Applying to repos
+
+When creating a new repo, please use any pre-commit hooks and tools that are useful.
+
+When working with an existing repo, please balance the benefits of adding a tool or pre-commit hook with considerations such as
+  - the amount of code churn
+  - how it will improve code maintainability
+  - the time it may add to CI runs.
+
+## Configuration of a tool
+
+Use your judgment as to whether to apply a strict or lenient configuration for linting code.
+Example configurations are in
+  - JupyterHub (A large repository which was converted and therefore has a fairly lax configuration)
+  - **TODO: Repository where pre-commit/flake8 was enforced near the start so has a strict configuration**
```

Comment:
It isn't enforced near start, but it is quite strict currently. It didn't cause much trouble to make the change as there were no other open PRs to cause conflict against.

See https://github.com/jupyterhub/nativeauthenticator's [.pre-commit-config.yaml](https://github.com/jupyterhub/nativeauthenticator/blob/main/.pre-commit-config.yaml) and [.flake8](https://github.com/jupyterhub/nativeauthenticator/blob/main/.flake8) config.

Reply:
Added!